docs/launcher.md

```diff
@@ -0,0 +1,90 @@
+
+# vLLM Launcher
+
+## Description
+
+This document describes a vLLM launcher that can be used to achieve model swapping without changes to vLLM. The launcher initializes and manages a subsidiary inference vLLM server and responses with the PID. The launcher does basic vLLM code loading and initialization work (like import modules) of the inference server so that this work does not have to be done at the startup of the inference server process, reducing the startup latency. In future work, the launcher could be extended to sleep and wake up the inference servers that have been initialize by its POST request, and support mutiple instances.
+
+## Launcher Methods and functionalities
+
+The client/user will use a launcher-specific command to run the launcher. To swap a model in, the Client will issue a POST request (to the launcher) that includes the model reference and the model-specific flags. The launcher sends an `OK` response to the client/controller to indicate that queries can be sent to the vLLM instance directly.
+
+Setting up process to create a new vLLM inference instance using POST:
+
+```mermaid
+sequenceDiagram
+    participant Client
+    participant Launcher
+    participant vLLM instance HTTP server
+
+    Client->>Launcher: POST create new instance
+    Launcher-->>vLLM instance HTTP server: create vLLM instance
+    Launcher->>Client: HTTP 200 OK, Data
+    Client->>vLLM instance HTTP server: POST Request
+    vLLM instance HTTP server->>Client: JSON text-generated response
+```
+
+An example of a CURL command that could be sent to the launcher can be seen below:
+
+```bash
+curl -X POST \
+  -H "Content-Type: application/json" \
+  -d 'JSON_EXAMPLE' \
+  http://localhost:8000/v1/vllm
+```
+
+where the `JSON_EXAMPLE` is:
+
+```json
+{
+  "options": "--model TinyLlama/TinyLlama-1.1B-Chat-v1.0 --port 8005",
+  "env_var": {
```

Comment:
This field name is singular, but the contents are plural. I would name this "env" or "env_vars".

Reply:
Why not update it in this PR?

Reply:
This doc should be updated too

Reply:
I saw and merged the PR to the code. This thread is not about the code, it is about the documentation.